APIAlfresco.php

```diff
@@ -79,13 +93,16 @@ public function checkResponse()
      */
     public function setFolderByPath($folder, array $options = array())
     {
-        $obj = $this->repository->getObjectByPath($folder, $options);
-        $propiedad = $obj->properties['cmis:baseTypeId'];
-        if ($propiedad != 'cmis:folder') {
-            echo 'The object is not a folder';
-            exit(255);
-        } else {
-            $this->parentFolder = $obj;
+        try {
+            $obj = $this->repository->getObjectByPath($folder, $options);
+            $properties = $obj->properties['cmis:baseTypeId'];
+            if ($properties != 'cmis:folder') {
+                throw new Exception('The object is not a folder');
+            } else {
```

Comment:
This `else` will never executed if the object is not a folder. So is redundant, instead of that you can create the flow like this:

```
// Validate the needed logic for making the normal processing before everything
if (missing value)
{
	throw new BadMethodCallException;
}

if (invalid argument)
{
	throw new InvalidArgumentException;
}

if (!bad processing at runtime)
{
	throw new RuntimeException;
}

....

// Everything is valid at this point, make good code here.
```